osu.Game.Rulesets.Osu.Tests/Mods/TestSceneOsuModHidden.cs

```diff
@@ -15,10 +15,7 @@ namespace osu.Game.Rulesets.Osu.Tests.Mods
 {
     public class TestSceneOsuModHidden : ModTestScene
     {
-        public TestSceneOsuModHidden()
-            : base(new OsuRuleset())
-        {
-        }
+        protected override Ruleset CreatePlayerRuleset() => new OsuRuleset();
```

Comment:
At the point where you're making an `TestSceneOsuPlayer` and deriving from it for other tests, you might also make an abstract `OsuModTestScene` and derive that for all the mod test scenes.

It'd probably be a good path forward to provide ready-made abstractions for all ruleset-requiring tests anyway in the template projects.

Reply:
> At the point where you're making an `TestSceneOsuPlayer` and deriving from it for other tests, you might also make an abstract `OsuModTestScene` and derive that for all the mod test scenes.
> 
> It'd probably be a good path forward to provide ready-made abstractions for all ruleset-requiring tests anyway in the template projects.

I was aware initially of making that for the `ModTestScene` in particular as that may potentially start containing mod-specific test scenes like the existing `ModPerfectTestScene`.

But actually that may help bringing up the ruleset used for the test scene to become more of a test-declaration code, should I enforce these ready-made abstraction classes for *all* usages? (excluding mod-specific test scenes)

Reply:
Applied only to rulesets that have more than one sub-`ModTestScene` (osu! ruleset currently).